### ISSUE TITLE
A lot of false notifications when switching desktops

### ISSUE THREAD

**Issue by savbran** [ViRb3/SylphyHornEx#7]
Hi, 
I'm using last version of SylphyHornEx (v.5.4.4) and each time I switch desktops (LetfCTRL + Win + Right/Left Arrow) I see in the Windows taskbar a lot of task notifications (blinking notifications), they are false notifications because nothing is happening on them. 
This is confusing because you can't see which task has really notified something, see image enclosured.
![false-notifications](https://user-images.githubusercontent.com/3647689/82735412-dfa78200-9d21-11ea-9cc7-10de7a994ecb.png)

**Comment by ViRb3** [comment]
Apologies for missing this issue, apparently by default GitHub disables notifications for forked repositories. I have now enabled them for the time being.

I have also noticed this issue and I am not sure why it is happening. Perhaps @kuukunen has an idea?

**Comment by ViRb3** [comment]
It is worth pointing out that for me the issue only happens when I use "go to virtual desktop number" hotkey, e.g. __Ctrl + Win + 1/2__. I suspect it could be a problem with focus.

**Comment by pvl-rodionov** [comment]
I have such an issue too. A false orange notification appears when SylphyHornEx fails to properly identify active window on the current desktop.
Stably I can obtain the situation by opening Explorer1, Cmd(active) on Desktop1, Explorer2(active) on Desktop2 and starting switching between Desktop1 and Desktop2.
Issue comes then Explorer2 is not identified as active. Moreover, after that Explorer2 may be seen on Desktop1 in the taskbar previews.
For me it takes place with any hotkey, e.g. Ctrl+Win+1/2, Ctrl+Win+Left/Right, Ctrl+Win+X.

**Comment by ViRb3** [comment]
@savbran @pvl-rodionov could you please try disabling the notification window (_Settings > Notification > Display a notification ..._) and seeing if you can reproduce the issue?

EDIT: The first time I tried this I couldn't reproduce the problem anymore, but now 5 minutes later it has come back.

**Comment by pvl-rodionov** [comment]
Confirm.

**Comment by ViRb3** [comment]
This is an interesting issue as SylphyHornEx doesn't do anything with (active) windows, it merely invokes the Windows API to switch virtual desktops. I will investigate this further tonight.

**Comment by pvl-rodionov** [comment]
Turned out that disabling _Skip default animation ..._ option solves the problem.

**Comment by ViRb3** [comment]
I have had that disabled but I still come across this issue when using non-standard keybinds (everything that is not __Ctrl + Win + Left/Right__). Could you try re-producing the issue with that option disabled and by pressing __Ctrl + Win + 1/2/etc__?

**Comment by pvl-rodionov** [comment]
Yeah, you are right, Ctrl+Win+1/2, Ctrl+Win+X still cause the false notification.

**Comment by ViRb3** [comment]
That's a good start, we know the problem is in how we switch virtual desktops and that we're not doing it the Microsoft way.

**Comment by mmrsva** [comment]
I noticed this issue only happens if there's a windows focused. Try to click on the taskbar (to leave focus from other windows) and then this doesn't happen when switching. If you click on a window before switching, the top most window on the switched desktop will blink. So it seems like a focus issue.

**Comment by Enteleform** [comment]
I've tried several virtual desktop management solutions and this seems to be a common issue. The only app I haven't experienced this issue with is zVirtualDesktop, here's a thread with a potential fix:

**[[zVirtualDesktop/#59] switching desktop causes focused window to blink in taskbar
](https://github.com/mzomparelli/zVirtualDesktop/issues/59)**

**Comment by Enteleform** [comment]
@ViRb3 just following up since you mentioned earlier in this thread that you had some issues w/ GitHub notifications.

Does the link in my previous comment seem like it could lead to resolution of this issue?

**Comment by ViRb3** [comment]
Hey, thanks for the link @Enteleform. I did indeed check it out, but I haven't had the time to test anything just yet. Hopefully I'll find some time over Christmas and report back here.

**Comment by ghost** [comment]
Would be amazing to get this fixed!

**Comment by florianehmke** [comment]
This is a showstopper for me, too distracting :(
